steel/src/lib.rs

```diff
@@ -13,6 +13,7 @@
 // limitations under the License.
 
 #![cfg_attr(not(doctest), doc = include_str!("../README.md"))]
+#![deny(rustdoc::broken_intra_doc_links)]
```

Comment:
Apparently, in the current stable this broken links detection does not work correctly, when the link is behind a feature.
This is why the doc test currently fails (https://github.com/risc0/risc0-ethereum/actions/runs/12003195929/job/33486808951?pr=336) while the docs-rs test is passing.
Either we have to remove this deny or remove the docs CI test and make sure everything is covered in the docs-rs test